python312Packages.svgdigitizer: init at 0.12.6

doronbehar opened the pull request:
## Description of changes


## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

doronbehar committed:
python312Packages.svgdigitizer: init at 0.12.6

doronbehar committed:
python312Packages.svgpathtools: init at 1.7.1

zhaofengli commented on pkgs/development/python-modules/svgdigitizer/default.nix:
```diff
@@ -0,0 +1,66 @@
+    svgwrite
+  ];
+
+  # no tests
+  doCheck = false;
```
This runs all sample code in the docstrings. There are a whole bunch of [other tests](https://github.com/echemdb/svgdigitizer/blob/4cdbb8e076aedd21e9ffb4c8ebed5419b7b23040/.github/workflows/test.yml) but they may be cumbersome to run. I think running the doctests is good enough.

```suggestion
  nativeCheckInputs = [
    pytestCheckHook
  ];

  pytestFlags = [
    "--doctest-modules"
    "svgdigitizer"
  ];
```

zhaofengli reviewed (changes requested):
Looks good, apart from the suggestion to run tests. Also upstream released 0.12.6 last week but it depends on `pybtex==0.25` (#421011, looks annoying 😞).

ethancedwards8 reviewed (approved):
lgtm with suggestsions by zhaofengli

doronbehar committed:
python313Packages.pybtex: 0.24.0 -> 0.25.1

doronbehar committed:
python312Packages.pybtex: no with lib; in meta

doronbehar committed:
python312Packages.pybtex: add meta.changelog

doronbehar committed:
python312Packages.pybtex: reorder inputs and arguments modernly

doronbehar committed:
python313Packages.pybtex: enable tests

doronbehar committed:
python312Packages.sphinxcontrib-bibtex: 2.6.3 -> 2.6.5

Diff: https://github.com/mcmtroffaes/sphinxcontrib-bibtex/compare/refs/tags/2.6.3...refs/tags/2.6.5

NOTE: This update is coupled to `pybtex: 0.24.0 -> 0.25.1` update from a
previous commit.

doronbehar commented:
> Also upstream released 0.12.6 last week but it depends on `pybtex==0.25` (#421011, looks annoying 😞).

Thanks for the link @zhaofengli ! It turned out to be not that hard :). Now this PR includes the necessary updates and hence supersedes https://github.com/NixOS/nixpkgs/pull/421011 .

doronbehar committed:
python312Packages.pybibget: use standard formatting

doronbehar committed:
python313Packages.pybibget: use setuptools in build-system

Needed ever since `pybtex` stopped propagating setuptools in a previous
commit.

doronbehar commented on pkgs/development/python-modules/svgdigitizer/default.nix:
```diff
@@ -0,0 +1,83 @@
+  matplotlib,
+  mergedeep,
+  pandas,
+  pdf2image,
```
> Looks like 0.12.6 [no longer](https://github.com/echemdb/svgdigitizer/blob/0.12.6/ChangeLog) depends on pdf2image (switched to pymupdf).

Wow thanks for the sharp review!

doronbehar committed:
python312Packages.pymatgen: disable another failing test on Darwin

doronbehar committed:
python313Packages.svgdigitizer: mark as broken on Darwin
